### PR TITLE
CIT-284 #time 2h fix access issue in backend and create JobDescription page

### PR DESCRIPTION
**Pull request by codal-kcolaco**

### CODE REVIEW

File: React-Projects/job-portal/src/Constants.jsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const POST_JOB_URL = "http://127.0.0.1:8000/api/employer/jobs/";
+export const GET_JOB_URL = "http://127.0.0.1:8000/api/applicants/jobs/";
+export const GET_SINGLE_JOB_URL = "http://127.0.0.1:8000/api/employer/jobs/";
+export const LOGIN_URL = "http://127.0.0.1:8000/api/login";
+export const SIGNUP_URL = "http://127.0.0.1:8000/api/register";
```
Comment (abhi-codal) [inline React-Projects/job-portal/src/Constants.jsx:5]: Please update this to something like below:
```

export const BE_ENDPOINT = "http://127.0.0.1:8000/api"
export const POST_JOB_URL = `${BE_ENDPOINT}/employer/jobs/`;
export const GET_JOB_URL = `${BE_ENDPOINT}/applicants/jobs/`;
......so on

```

File: React-Projects/job-portal/src/api.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+} from "./Constants";
+
+export const registerUser = async (fullname, email, password) => {
+  const requestBody = {
+    name: fullname,
+    email: email,
+    password: password,
+  };
+
+  const response = await fetch(SIGNUP_URL, {
```
Comment (abhi-codal) [inline React-Projects/job-portal/src/api.js:16]: @codal-kcolaco Please update this API call to axios.

File: React-Projects/job-portal/src/api.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  return response.json();
+};
+
+export const loginUser = async (email, password) => {
+  const requestBody = {
+    email: email,
+    password: password,
+  };
+
+  const response = await fetch(LOGIN_URL, {
```
Comment (abhi-codal) [inline React-Projects/job-portal/src/api.js:44]: @codal-kcolaco Please update this API call to axios.

File: React-Projects/job-portal/src/api.js (hunk 2 above)
Comment (codal-kcolaco) [reply React-Projects/job-portal/src/api.js:16]: @abhi-codal Understood

File: React-Projects/job-portal/src/api.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+  if (!response.ok) {
+    throw new Error("Network response is not ok");
+  }
+
+  return response.json();
+};
+
+export const fetchJobs = async () => {
+  const response = await fetch(GET_JOB_URL, {
```
Comment (abhi-codal) [inline React-Projects/job-portal/src/api.js:97]: @codal-kcolaco Please update all this API call to axios below.

File: React-Projects/job-portal/src/styles/Header.scss [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    .sign-up-button:hover,
+    .login-button:hover {
+      opacity: 0.3;
+    }
+  }
+}
+
+nav ul {
```
Comment (abhi-codal) [inline React-Projects/job-portal/src/styles/Header.scss:56]: @codal-kcolaco Please avoid giving styles directly to tags. always use classname to it.

Comment (codal-kcolaco) [reply React-Projects/job-portal/src/styles/Header.scss:56]: @abhi-codal Will make sure from now on

**Review by abhi-codal — APPROVED** [review]

**Pull request merged**